components/net/indexeddb/engines/sqlite/serialize.rs

```diff
@@ -0,0 +1,360 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+use net_traits::indexeddb_thread::IndexedDBKeyType;
+
+/// A module for encoding and decoding various IndexedDBKeyType types according to the basic strategy:
+///
+/// - Numbers:  0x10 + 8-byte sortable float
+/// - Dates:    0x20 + 8-byte sortable float
+/// - Strings:  0x30 + encoded chars + 0 terminator
+/// - Binaries: 0x40 + encoded octets + 0 terminator
+/// - Arrays:   0x50 + encoded items + 0 terminator
```

Comment:
Can we do:
```rust
enum SerializedValuePrefix {
    Number = 0x10,
    Date = 0x20,
    String = 0x30,
    Binary = 0x40,
    Array = 0x50,
}
```
and then refer to those values instead of using magic constants in this file?